pkg/server/router/merge/merge_openapi_v2.go

```diff
@@ -0,0 +1,142 @@
+package merge
+
+import (
+	"fmt"
+	"slices"
+
+	openapiv2 "github.com/google/gnostic-models/openapiv2"
+	"sigs.k8s.io/yaml"
+)
+
+func mergeSlicesOnValue[T comparable, C comparable](lhs, rhs []T, getter func(T) C) []T {
+	if len(lhs) == 0 {
+		return rhs
+	}
+
+	if len(rhs) == 0 {
+		return lhs
+	}
+
+	for _, t := range lhs {
```

Comment:
Shouldn't this be `for _, t := range rhs {` , leaving the rest of the loop as is? Otherwise it needs an explanatory comment. Right now this loop does nothing (unless `getter(x) => rand.Int()`)

Reply:
Yep, good catch thanks!